src/engine.rs

```diff
@@ -765,6 +762,9 @@ impl Reedline {
             for event in reedline_events.drain(..) {
                 match self.handle_event(prompt, event)? {
                     EventStatus::Exits(signal) => {
+                        if self.use_kitty_protocol {
+                            let _ = execute!(io::stdout(), event::PopKeyboardEnhancementFlags);
+                        }
```

Comment:
This is probably the wrong place to handle this, we have the split into `read_line` and `read_line_helper` to make sure that we disable extended terminal modes when leaving `read_line` independent of whether it succeeded or exited with an `Err`.

Reply:
Got it, I'll move out from the `_helper` as is, since the pop operation is perceived as infallible.